source/backend/dal/Repositories/ActivityRepository.cs

```diff
@@ -91,14 +93,34 @@ public PimsActivityInstance Add(PimsActivityInstance instance)
             return instance;
         }
 
-        public PimsActivityInstance Update(PimsActivityInstance instance)
+        public Tuple<PimsActivityInstance, PimsActivityInstanceNote> Update(PimsActivityInstance instance)
```

Comment:
yeah, I'm not a big fan of this. I think the service should be responsible for updating the activity and then also adding a note (in one transaction).

Creating repository methods that combine operations like this mean we can't compose individual operations anymore. Ie: it is no longer possible to update an activity without generating a note.

Also in general repositories should not affect external tables.

Thankfully this shouldn't be a big change. There is already a repository to add a note, so you can just revert this method and make the service responsible for composing the update activity and add note repository calls.

Reply:
The repository is only responsible for providing a note instance as the decision to whether create a note or not depends on the current activity status and updated activity status which is only available in  repository. 
In the code the service is only responsible for updating the activity and is also adding a note (in one transaction) using the entity note repo.

Please check the ActivityService code and let me know if still needs changes.

Reply:
changed as per feedback.